File: app/src/main/java/com/sharpdroid/registroelettronico/Views/CSwipeRefreshLayout.java
package com.sharpdroid.registroelettronico.Views;

import android.content.Context;
import android.support.v4.widget.SwipeRefreshLayout;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.ViewConfiguration;

public class CSwipeRefreshLayout extends SwipeRefreshLayout {
    private int mTouchSlop;
    private float mPrevx;
    private float mPrevy;

    public CSwipeRefreshLayout(Context context) {
        this(context, null);
    }

    public CSwipeRefreshLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
        mTouchSlop = ViewConfiguration.get(context).getScaledTouchSlop();
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        switch (ev.getAction()) {
            case MotionEvent.ACTION_DOWN:
                mPrevx = MotionEvent.obtain(ev).getX();
                mPrevy = MotionEvent.obtain(ev).getY();
                break;
            case MotionEvent.ACTION_MOVE:
                final float evX = ev.getX();
                final float evy = ev.getY();
                float xDiff = Math.abs(evX - mPrevx);
                float yDiff = Math.abs(evy - mPrevy);
                if (xDiff > mTouchSlop && xDiff > yDiff) {
                    return false;
                }
        }
        return super.onInterceptTouchEvent(ev);
    }
}

File: app/src/main/java/com/sharpdroid/registroelettronico/views/CSwipeRefreshLayout.kt
package com.sharpdroid.registroelettronico.views

import android.content.Context
import android.support.v4.widget.SwipeRefreshLayout
import android.util.AttributeSet
import android.view.MotionEvent
import android.view.ViewConfiguration

class CSwipeRefreshLayout(context: Context, attrs: AttributeSet? = null) : SwipeRefreshLayout(context, attrs) {
    private val mTouchSlop: Int = ViewConfiguration.get(context).scaledTouchSlop
    private var mPrevX: Float = 0f
    private var mPrevY: Float = 0f

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        when (ev.action) {
            MotionEvent.ACTION_DOWN -> {
                mPrevX = MotionEvent.obtain(ev).x
                mPrevY = MotionEvent.obtain(ev).y
            }
            MotionEvent.ACTION_MOVE -> {
                val evX = ev.x
                val evy = ev.y
                val xDiff = Math.abs(evX - mPrevX)
                val yDiff = Math.abs(evy - mPrevY)
                if (xDiff > mTouchSlop && xDiff > yDiff) {
                    return false
                }
            }
        }
        return super.onInterceptTouchEvent(ev)
    }
}
